### ISSUE TITLE
Lexer initialization errors are not reported

### ISSUE THREAD

**Issue by maelmahdy** [b3b00/csly#186]
Thanks for this really promising project, it's very nice idea

I was playing with it today and started by declaring the Terminals enum and I used both regex and generic lexemes in the same enum then tried to parse an expression but it throws NullReferenceException

I noticed that the parser.Lexer is null but didn't find any way to get the initialization error so I'd to dig down into the code until I found this block in the LexerBuilder class:
```
if (hasGenericLexemes && hasRegexLexemes)
{
    result.AddError(new LexerInitializationError(ErrorLevel.WARN,
    "cannot mix Regex lexemes and Generic lexemes in same lexer"));
}
```
this error statement is very obvious but there's no way to get it by the class end user because you're calling the method as below:
```
var lexerResult = BuildLexer(extensionBuilder);
parser.Lexer = lexerResult.Result;
```
the lexerResult itself is not saved anywhere so we can't access lexerResult.Errors

It'll be very useful to expose it as public member so we can know why the lexer was not initialized.

Thanks,

**Comment by b3b00** [comment]
Yes indeed this is a real issue. if BuildLexer returns an error thant full build must stop since parser will not be able to get a token stream later. fixing it right now

**Comment by b3b00** [comment]
issue fixed and  nuget 2.6.0.2 published
